src/utils/stackedUtils.ts

```diff
@@ -0,0 +1,179 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+  export module Utils {
+    type StackedDatum = {
+      key: any;
+      value: number;
+      offset?: number;
+    };
+
+    export class Stacked {
+
+      /**
+       * Calculates the offset of each piece of data, in each dataset, relative to the baseline,
+       * for drawing purposes.
+       *
+       * @return {Utils.Map<Dataset, D3.Map<number>>} A map from each dataset to the offset of each datapoint
+       */
+      public static computeStackOffsets(datasets: Dataset[], keyAccessor: Accessor<any>, valueAccessor: Accessor<any>) {
+        var domainKeys = this.getDomainKeys(datasets, keyAccessor);
+
+        var dataMapArray = this._generateDefaultMapArray(datasets, keyAccessor, valueAccessor, domainKeys);
+
+        var positiveDataMapArray: D3.Map<StackedDatum>[] = dataMapArray.map((dataMap) => {
+          return Utils.Methods.populateMap(domainKeys, (domainKey) => {
+            return { key: domainKey, value: Math.max(0, dataMap.get(domainKey).value) || 0 };
+          });
+        });
+
+        var negativeDataMapArray: D3.Map<StackedDatum>[] = dataMapArray.map((dataMap) => {
+          return Utils.Methods.populateMap(domainKeys, (domainKey) => {
+            return { key: domainKey, value: Math.min(dataMap.get(domainKey).value, 0) || 0 };
+          });
+        });
+
+        var stackOffsets = this._generateStackOffsets(
```

Comment:
How does the `this` keyword work under this circumstance?  Shouldn't the writing be `Stacked._generateStackOffsets(...)`